lte/gateway/c/li_agent/main.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <lte/protos/mconfig/mconfigs.pb.h>
+#include <thread>
+
+#include "MagmaService.h"
+#include "MConfigLoader.h"
+#include "ServiceRegistrySingleton.h"
+
+#include "InterfaceMonitor.h"
+#include "PDUGenerator.h"
+#include "ProxyConnector.h"
+
+#define LIAGENTD "liagentd"
+#define LIAGENTD_VERSION "1.0"
+
+static magma::mconfig::LIAgentD get_default_mconfig() {
+  magma::mconfig::LIAgentD mconfig;
+  mconfig.set_log_level(magma::orc8r::LogLevel::INFO);
+  return mconfig;
+}
+
+static magma::mconfig::LIAgentD load_mconfig() {
+  magma::mconfig::LIAgentD mconfig;
+  magma::MConfigLoader loader;
+  if (!loader.load_service_mconfig(LIAGENTD, &mconfig)) {
+    MLOG(MERROR) << "Unable to load mconfig for liagentd, using default";
+    return get_default_mconfig();
+  }
+  return mconfig;
+}
+
+static uint32_t get_log_verbosity(
+    const YAML::Node& config, magma::mconfig::LIAgentD mconfig) {
+  if (!config["log_level"].IsDefined()) {
+    if (mconfig.log_level() < 0 || mconfig.log_level() > 4) {
+      return MINFO;
+    }
+    return mconfig.log_level();
+  }
+  std::string log_level = config["log_level"].as<std::string>();
+  if (log_level == "DEBUG") {
+    return MDEBUG;
+  } else if (log_level == "INFO") {
+    return MINFO;
+  } else if (log_level == "WARNING") {
+    return MWARNING;
+  } else if (log_level == "ERROR") {
+    return MERROR;
+  } else if (log_level == "FATAL") {
+    return MFATAL;
+  } else {
+    MLOG(MINFO) << "Invalid log level in config: "
+                << config["log_level"].as<std::string>();
+    return MINFO;
+  }
+}
+
+int main(void) {
+  magma::init_logging(LIAGENTD);
+
+  auto mconfig = load_mconfig();
+  auto config  = magma::ServiceConfigLoader{}.load_service_config(LIAGENTD);
+  magma::set_verbosity(get_log_verbosity(config, mconfig));
+  MLOG(MINFO) << "Starting LI Agent service";
+
+  // TODO: Ignore SigPipe as SSL_write throws that occasionally, investigate
```

Comment:
Can we create a signal handler that logs when we do get SigPipe? So that it's visible when this is firing... Or is there another log tell beyond this now-disabled sig?

Reply:
Yeah thats what I had originally but for some reason it still recieved SIGPIPE (not in signal handler). I'm assuming because it doesn't propagate properly to the callback function.
LMK if you have any ideas